runtime/oti/j9.h

```diff
@@ -325,7 +325,9 @@ static const struct { \
 
 /* Macros for ValueTypes */
 #ifdef J9VM_OPT_VALHALLA_VALUE_TYPES
-#define J9_IS_J9CLASS_VALUETYPE(clazz) J9_ARE_ALL_BITS_SET(clazz->classFlags, J9ClassIsValueType)
+#define J9_IS_J9CLASS_VALUETYPE(clazz) J9_ARE_ALL_BITS_SET((clazz)->classFlags, J9ClassIsValueType)
```

Comment:
you'll need to add the macros `#else /* J9VM_OPT_VALHALLA_VALUE_TYPES */` case as well

Reply:
Would this work @tajila :

```#define J9_IS_J9CLASS_FLATTENED(clazz) FALSE
#define J9_VALUETYPE_FLATTENED_SIZE(clazz)(0)```

Reply:
Not too sure on what to define  `J9_VALUETYPE_FLATTENED_SIZE` as, but maybe we won't even need it if `J9_IS_J9CLASS_FLATTENED(clazz)` is false

Reply:
change J9_VALUETYPE_FLATTENED_SIZE(clazz) to `((UDATA) -1)`.

Reply:
UDATA_MAX might be better than the cast value.

Reply:
I see the value is still 0 - would it be better to simply not define the macro if VTs are not in the compile? The "is flat" macro can stay, but the size macro should go.

Reply:
I think I had a compile error when the size macro was not defined... I'll try to reproduce it and paste the error here